### PR TITLE
[UIKit] Update bindings to Xcode 13 Beta 5

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: src/uikit.cs [hunk 1]
```diff
@@ -3798,6 +3898,14 @@ interface UICollectionViewDelegate : UIScrollViewDelegate {
 		[return: NullAllowed]
 		NSIndexPath GetIndexPathForPreferredFocusedView (UICollectionView collectionView);
 
+		[NoWatch, NoTV, iOS (15,0), MacCatalyst (15,0)]
+		[Export ("collectionView:selectionFollowsFocusForItemAtIndexPath:")]
+		bool GetSelectionFollowsFocusForItemAtIndexPath (UICollectionView collectionView, NSIndexPath indexPath);
```
Comment (rolfbjarne) [inline src/uikit.cs:3903]:
```suggestion
		bool GetSelectionFollowsFocusForItem (UICollectionView collectionView, NSIndexPath indexPath);
```

File: src/uikit.cs [hunk 2]
```diff
@@ -7986,35 +8207,81 @@ interface UIButton : UIAccessibilityContentSizeCategoryImageAdjusting
...
 		UIButton FromType (UIButtonType type);
 
+		[Deprecated (PlatformName.iOS, 15, 0, message: "Ignored when 'UIButtonConfiguration' is used.")]
+		[Deprecated (PlatformName.TvOS, 15, 0, message: "Ignored when 'UIButtonConfiguration' is used.")]
 		[Appearance]
 		[Export ("contentEdgeInsets")]
 		UIEdgeInsets ContentEdgeInsets {get;set;}
 
+		[Deprecated (PlatformName.iOS, 15, 0, message: "Ignored when 'UIButtonConfiguration' is used.")]
+		[Deprecated (PlatformName.TvOS, 15, 0, message: "Ignored when 'UIButtonConfiguration' is used.")]
```
Comment (rolfbjarne) [inline src/uikit.cs:8245]: It's not deprecated on the watch?

File: src/uikit.cs [hunk 3]
```diff
@@ -8095,15 +8366,27 @@ interface UIButton : UIAccessibilityContentSizeCategoryImageAdjusting
 		[Export ("imageView", ArgumentSemantic.Retain)]
 		UIImageView ImageView { get; }
 
+		[TV (15,0), Watch (8,0), iOS (15,0), MacCatalyst (15,0)]
+		[NullAllowed, Export ("subtitleLabel", ArgumentSemantic.Strong)]
+		UILabel SubtitleLabel { get; }
+
+		[Deprecated (PlatformName.iOS, 15, 0, message: "Ignored when 'Configuration' is used, override 'LayoutSubviews', call base, and set position views on your own.")]
+		[Deprecated (PlatformName.TvOS, 15, 0, message: "Ignored when 'Configuration' is used, override 'LayoutSubviews', call base, and set position views on your own.")]
```
Comment (rolfbjarne) [inline src/uikit.cs:8374]: Same: not deprecated on the watch?

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/uikit.cs (hunk 2 above)
Comment (dalexsoto) [reply src/uikit.cs:8245]: This type is not available on Watch

```
#if !WATCH
	[BaseType (typeof (UIControl))]
	interface UIButton : UIAccessibilityContentSizeCategoryImageAdjusting
#if IOS
```

File: src/uikit.cs (hunk 3 above)
Comment (dalexsoto) [reply src/uikit.cs:8374]: This type is not available on Watch

```
#if !WATCH
	[BaseType (typeof (UIControl))]
	interface UIButton : UIAccessibilityContentSizeCategoryImageAdjusting
#if IOS
```

**Comment by dalexsoto** [comment]
@rolfbjarne feedback applied, thank you!

File: src/uikit.cs (hunk 2 above)
Comment (rolfbjarne) [reply src/uikit.cs:8245]: Then the [NoWatch] attribute on the other members added to this type is redundant.

File: src/uikit.cs (hunk 3 above)
Comment (rolfbjarne) [reply src/uikit.cs:8374]: There's am `[Watch (8, 0)]` on the previous member which should be removed then, at best it's confusing.

**Review by rolfbjarne — APPROVED** [review]

File: src/uikit.cs [hunk 4]
```diff
@@ -3109,6 +3183,10 @@ interface UIApplicationDelegate {
 		[iOS (13,0), TV (13,0), Watch (6,0)]
 		[Export ("application:didDiscardSceneSessions:")]
 		void DidDiscardSceneSessions (UIApplication application, NSSet<UISceneSession> sceneSessions);
+
+		[iOS (15,0), TV (15,0), Watch (8,0), MacCatalyst (15,0)]
+		[Export ("applicationShouldAutomaticallyLocalizeKeyCommands:")]
+		bool GetShouldAutomaticallyLocalizeKeyCommands (UIApplication application);
```
Comment (spouliot) [inline src/uikit.cs:3189]: not sure the `Get` is needed here, `Should` is a verb

File: src/uikit.cs (hunk 2 above)
Comment (spouliot) [reply src/uikit.cs:8245]: some members, a few lines down, are decorated `[Watch (8,0)]`

File: src/uikit.cs [hunk 5]
```diff
@@ -8955,6 +9256,11 @@ interface UINavigationBar : UIBarPositioning, NSCoding {
 		[NullAllowed, Export ("scrollEdgeAppearance", ArgumentSemantic.Copy)]
 		UINavigationBarAppearance ScrollEdgeAppearance { get; set; }
 
+		[Appearance]
+		[TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+		[NullAllowed, Export ("compactScrollEdgeAppearance", ArgumentSemantic.Copy)]
+		UINavigationBarAppearance compactScrollEdgeAppearance { get; set; }
```
Comment (spouliot) [inline src/uikit.cs:9262]: typo: `compactScrollEdgeAppearance` needs a capital `C`

File: src/uikit.cs [hunk 6]
```diff
@@ -9108,6 +9414,10 @@ interface UINavigationItem : NSCoding {
 		[TV (13,0), iOS (13,0)]
 		[NullAllowed, Export ("scrollEdgeAppearance", ArgumentSemantic.Copy)]
 		UINavigationBarAppearance ScrollEdgeAppearance { get; set; }
+
+		[TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+		[NullAllowed, Export ("compactScrollEdgeAppearance", ArgumentSemantic.Copy)]
+		UINavigationBarAppearance compactScrollEdgeAppearance { get; set; }
```
Comment (spouliot) [inline src/uikit.cs:9420]: same, needs capital `C`

File: src/uikit.cs [hunk 7]
```diff
@@ -20822,6 +21414,11 @@ interface UIPointerRegion : NSCopying {
 	[DisableDefaultCtor]
 	interface UIPointerStyle : NSCopying {
 
+		// TODO: Enable with the UIPointerAccessory bindings
```
Comment (spouliot) [inline src/uikit.cs:21417]: ? forgotten or does it need a tracking issue ?

File: src/uikit.cs [hunk 8]
```diff
@@ -22133,4 +22775,348 @@ interface UIPrinterDestination : NSSecureCoding {
...
+	interface UIFocusEffect : NSCopying {
+
+		[Static]
+		[Export ("effect")]
+		UIFocusEffect Create ();
+	}
+
+	[NoWatch, NoTV, iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (UIFocusEffect))]
+	interface UIFocusHaloEffect {
```
Comment (spouliot) [inline src/uikit.cs:22944]: looks like a `[DisableDefaultCtor]` should be present here too

**Review by spouliot — CHANGES_REQUESTED** [review]

**Comment by dalexsoto** [comment]
@spouliot Applied feedback! Also added `UIPointerAccessory` bindings

File: tests/monotouch-test/UIKit/UIPointerAccessory.cs [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		[Test]
+		public void UIPointerAccessoryPositionTopTest ()
+		{
+			UIPointerAccessory acc = null;
+			Assert.DoesNotThrow (() => acc = UIPointerAccessory.CreateArrow (UIPointerAccessoryPosition.Top), "Should not throw");
+			Assert.NotNull (acc, $"{nameof (acc)} was null");
+			Assert.AreEqual (acc.Position.Offset, UIPointerAccessoryPosition.Top.Offset, "Offset");
+			Assert.AreEqual (acc.Position.Angle, UIPointerAccessoryPosition.Top.Angle, "Offset");
```
Comment (spouliot) [inline tests/monotouch-test/UIKit/UIPointerAccessory.cs:36]: minor: "Angle"

File: tests/monotouch-test/UIKit/UIPointerAccessory.cs [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		[Test]
+		public void UIPointerAccessoryPositionTopRightTest ()
+		{
+			UIPointerAccessory acc = null;
+			Assert.DoesNotThrow (() => acc = UIPointerAccessory.CreateArrow (UIPointerAccessoryPosition.TopRight), "Should not throw");
+			Assert.NotNull (acc, $"{nameof (acc)} was null");
+			Assert.AreEqual (acc.Position.Offset, UIPointerAccessoryPosition.TopRight.Offset, "Offset");
+			Assert.AreEqual (acc.Position.Angle, UIPointerAccessoryPosition.TopRight.Angle, "Offset");
```
Comment (spouliot) [inline tests/monotouch-test/UIKit/UIPointerAccessory.cs:46]: minor: "Angle"

File: tests/monotouch-test/UIKit/UIPointerAccessory.cs [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		[Test]
+		public void UIPointerAccessoryPositionRightTest ()
+		{
+			UIPointerAccessory acc = null;
+			Assert.DoesNotThrow (() => acc = UIPointerAccessory.CreateArrow (UIPointerAccessoryPosition.Right), "Should not throw");
+			Assert.NotNull (acc, $"{nameof (acc)} was null");
+			Assert.AreEqual (acc.Position.Offset, UIPointerAccessoryPosition.Right.Offset, "Offset");
+			Assert.AreEqual (acc.Position.Angle, UIPointerAccessoryPosition.Right.Angle, "Offset");
```
Comment (spouliot) [inline tests/monotouch-test/UIKit/UIPointerAccessory.cs:56]: minor: "Angle"

File: tests/monotouch-test/UIKit/UIPointerAccessory.cs [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		[Test]
+		public void UIPointerAccessoryPositionBottomRightTest ()
+		{
+			UIPointerAccessory acc = null;
+			Assert.DoesNotThrow (() => acc = UIPointerAccessory.CreateArrow (UIPointerAccessoryPosition.BottomRight), "Should not throw");
+			Assert.NotNull (acc, $"{nameof (acc)} was null");
+			Assert.AreEqual (acc.Position.Offset, UIPointerAccessoryPosition.BottomRight.Offset, "Offset");
+			Assert.AreEqual (acc.Position.Angle, UIPointerAccessoryPosition.BottomRight.Angle, "Offset");
```
Comment (spouliot) [inline tests/monotouch-test/UIKit/UIPointerAccessory.cs:66]: minor: "Angle"

File: tests/monotouch-test/UIKit/UIPointerAccessory.cs [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		[Test]
+		public void UIPointerAccessoryPositionBottomTest ()
+		{
+			UIPointerAccessory acc = null;
+			Assert.DoesNotThrow (() => acc = UIPointerAccessory.CreateArrow (UIPointerAccessoryPosition.Bottom), "Should not throw");
+			Assert.NotNull (acc, $"{nameof (acc)} was null");
+			Assert.AreEqual (acc.Position.Offset, UIPointerAccessoryPosition.Bottom.Offset, "Offset");
+			Assert.AreEqual (acc.Position.Angle, UIPointerAccessoryPosition.Bottom.Angle, "Offset");
```
Comment (spouliot) [inline tests/monotouch-test/UIKit/UIPointerAccessory.cs:76]: minor: "Angle"

File: tests/monotouch-test/UIKit/UIPointerAccessory.cs [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		[Test]
+		public void UIPointerAccessoryPositionBottomLeftTest ()
+		{
+			UIPointerAccessory acc = null;
+			Assert.DoesNotThrow (() => acc = UIPointerAccessory.CreateArrow (UIPointerAccessoryPosition.BottomLeft), "Should not throw");
+			Assert.NotNull (acc, $"{nameof (acc)} was null");
+			Assert.AreEqual (acc.Position.Offset, UIPointerAccessoryPosition.BottomLeft.Offset, "Offset");
+			Assert.AreEqual (acc.Position.Angle, UIPointerAccessoryPosition.BottomLeft.Angle, "Offset");
```
Comment (spouliot) [inline tests/monotouch-test/UIKit/UIPointerAccessory.cs:86]: minor: "Angle"

File: tests/monotouch-test/UIKit/UIPointerAccessory.cs [hunk 15]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		[Test]
+		public void UIPointerAccessoryPositionLeftTest ()
+		{
+			UIPointerAccessory acc = null;
+			Assert.DoesNotThrow (() => acc = UIPointerAccessory.CreateArrow (UIPointerAccessoryPosition.Left), "Should not throw");
+			Assert.NotNull (acc, $"{nameof (acc)} was null");
+			Assert.AreEqual (acc.Position.Offset, UIPointerAccessoryPosition.Left.Offset, "Offset");
+			Assert.AreEqual (acc.Position.Angle, UIPointerAccessoryPosition.Left.Angle, "Offset");
```
Comment (spouliot) [inline tests/monotouch-test/UIKit/UIPointerAccessory.cs:96]: minor: "Angle"

File: tests/monotouch-test/UIKit/UIPointerAccessory.cs [hunk 16]
```diff
@@ -0,0 +1,110 @@
...
+		}
+
+		[Test]
+		public void UIPointerAccessoryPositionTopLeftTest ()
+		{
+			UIPointerAccessory acc = null;
+			Assert.DoesNotThrow (() => acc = UIPointerAccessory.CreateArrow (UIPointerAccessoryPosition.TopLeft), "Should not throw");
+			Assert.NotNull (acc, $"{nameof (acc)} was null");
+			Assert.AreEqual (acc.Position.Offset, UIPointerAccessoryPosition.TopLeft.Offset, "Offset");
+			Assert.AreEqual (acc.Position.Angle, UIPointerAccessoryPosition.TopLeft.Angle, "Offset");
```
Comment (spouliot) [inline tests/monotouch-test/UIKit/UIPointerAccessory.cs:106]: minor: "Angle"

**Review by spouliot — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]
Needs some cleanup and tests.

File: src/UIKit/UITypes.cs [hunk 17]
```diff
@@ -150,7 +150,46 @@ public override int GetHashCode ()
...
+	public struct UIPointerAccessoryPosition {
+		public nfloat Offset, Angle;
+
+		public UIPointerAccessoryPosition (nfloat offset, nfloat angle)
+		{
+			Offset = offset;
+			Angle = angle;
+		}
+
+#if !COREBUILD
```
[the remark this answers is not in the document]
Comment (dalexsoto) [reply src/UIKit/UITypes.cs:171]: Huh? Tests are here ;) https://github.com/xamarin/xamarin-macios/pull/12706/files#diff-cb1e9a111d26dfb617f1a16d03684e3d364f4dfec4e4ba1b83738f59bfeb95dc

File: src/UIKit/UITypes.cs [hunk 18]
```diff
@@ -150,7 +150,46 @@ public override int GetHashCode ()
 		public static UIFloatRange Infinite = new UIFloatRange (nfloat.NegativeInfinity, nfloat.PositiveInfinity);
 	}
 #endif
-	
+
+#if IOS
```
[the remark this answers is not in the document]
Comment (dalexsoto) [reply src/UIKit/UITypes.cs:155]: mmm Availability is there too!

```csharp
#if !NET
 	[Introduced (PlatformName.iOS, 15,0)]
 #else
 	[SupportedOSPlatform ("ios15.0")]
 #endif //!NET
```

**Comment by dalexsoto** [comment]
Known unrelated issues.
* https://github.com/xamarin/maccore/issues/2414
* https://github.com/xamarin/maccore/issues/2379

**Pull request merged**